content/applications/inventory_and_mrp/barcode/setup/device_troubleshooting.rst

```diff
@@ -0,0 +1,214 @@
+the scanners do not work as intended, and Odoo returns errors to the device.
+
+Read the sections below to identify general and unique device issues, related to specific, popular
+types of scanners.
+
+.. seealso::
+   - :doc:`../setup/hardware`
+   - :doc:`../setup/software`
+
+General issues
+==============
+
+Refer to the following sections below for common issues involving popular barcode scanner devices.
+
+For issues related to specific devices, refer to the :ref:`Android scanners
+<barcode/setup/android-scanners>` section for mobile computer scanners, or to the
+:ref:`Screenless scanners <barcode/setup/screenless-scanners>` section for USB and bluetooth
+scanners.
+
+Barcode cannot be read
+----------------------
+
+One common issue encountered when using barcode scanners is an error resulting from barcodes that
+cannot be read.
+
+This can occur due to any of the following reasons:
+
+- The barcode is damaged.
+- The device cannot read the required barcode type (some scanners can only read 2D barcodes).
+- The barcode being scanned is on a screen. Some scanners don't support this, and the barcodes
+  **must** be printed out to be scanned. This is most common with 1D barcodes.
+- The device has no battery, or is broken. To rule this out, follow the troubleshooting instructions
+  in the following sections.
+
+Odoo returns barcode error
+--------------------------
+
+All types of barcode scanners have their own device "language", which affects how they output
+barcode data to Odoo's *Barcode* app. Sometimes, this can cause Odoo *Barcode* to return a barcode
+error after scanning. This could be due to any of the following reasons:
+
+- The computer is configured with a different keyboard than the barcode scanner. To rule this out,
+  ensure that the device is configured with the same keyboard.
+
+  For example, if the computer is configured to use an FR-BE keyboard, configure the scanner to send
+  FR-BE keystrokes. The same logic applies if using a tablet instead of a computer.
+
+  For more information on configuring keystrokes, refer to the :doc:`Barcode scanner setup
+  <hardware>` documentation.
+- For mobile computer scanners (such as, Zebra devices), the scanner might interpret the barcode
+  differently than intended. To rule this out, scan a test barcode to see how the scanner interprets
+  the barcode.
+
+.. _barcode/setup/android-scanners:
+
+Android scanners
+================
+
+The most recent barcode scanner models using Android and Google Chrome should work with Odoo.
+However, due to the variety of models and configurations, it is recommended to first test a
+scanner's compatibility with Odoo.
+
+The Zebra product line is recommended; specifically, the **Zebra TC21 (WiFi-only)**, and **Zebra
+TC26 (WiFi/cellular)**.
+
+.. seealso::
+   `Odoo Inventory & Barcode • Compatible Hardware <https://www.odoo.com/app/inventory-hardware>`_
```

Comment:
I prefer this formatting for the link, but if there is a reason to format it as "Odoo Inventory & Barcode • Compatible Hardware" that I am not aware of, please let me know! :)
```suggestion
   `Odoo Inventory & Barcode compatible hardware <https://www.odoo.com/app/inventory-hardware>`_
```

Reply:
No reason, I just liked the way it looked. Making this change - Suggestion Implemented ✅